server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweakBuffer.java

```diff
@@ -518,38 +462,50 @@ private int checkStateRight() {
 	// get the difference in feet position between the kinematic and corrected
 	// positions of the feet disregarding vertical displacement
 	private float getLeftFootHorizontalDifference() {
-		return leftFootPositionCorrected.subtract(leftFootPosition).setY(0).length();
+		return new Vector3(
+			leftFootPositionCorrected.minus(leftFootPosition).getX(),
+			0f,
+			leftFootPositionCorrected.minus(leftFootPosition).getZ()
+		).len();
 	}
 
 	// get the difference in feet position between the kinematic and corrected
 	// positions of the feet
 	private float getRightFootHorizontalDifference() {
-		return rightFootPositionCorrected.subtract(rightFootPosition).setY(0).length();
+		return new Vector3(
+			rightFootPositionCorrected.minus(rightFootPosition).getX(),
+			0f,
+			rightFootPositionCorrected.minus(rightFootPosition).getZ()
+		).len();
```

Comment:
Same as last comment, why are you calculating this twice?
```suggestion
		Vector3 diff = rightFootPositionCorrected.minus(rightFootPosition);
		return new Vector3(
			diff.getX(),
			0f,
			diff.getZ()
		).len();
```